Zend/zend_execute.c

```diff
@@ -1795,6 +1795,18 @@ static zend_never_inline void zend_fetch_dimension_address_UNSET(zval *result, z
 	zend_fetch_dimension_address(result, container_ptr, dim, dim_type, BP_VAR_UNSET);
 }
 
+static zend_always_inline int zend_fetch_dimension_address_read_valid_container(zval *container, int support_strings) {
+	if (EG(current_execute_data)->opline->opcode != ZEND_FETCH_LIST) {
+		if (EG(current_execute_data)->opline->op1_type != IS_VAR) {
```

Comment:
The check here for not IS_VAR is to aide in preventing raising multiple warnings for
```
$a = null;
$a[1][2][3];
```
When accessing $a[1], the container op1_type is a CV, but the return of that for access of [2] is VAR.  This was the attempt to only show the warning for user based code like this.

Although since you made me double check that I have found that it's not necessarily perfect.
```
function foo() {
    return null;
}
foo()[1];
```
Does not properly throw any warning, since the return of a function IS_VAR

I'll change it around to look if IS_VAR the previous op wasn't a function call, so we can properly throw warning on the first access post-call.